### ISSUE TITLE
`sf apex get test` throws heap out of memory

### ISSUE THREAD

**Issue by pawel-id** [forcedotcom/salesforcedx-vscode#5589]
### Summary

We have quite big project having currently c.a.  2300+ apex tests. Starting with version @salesforce/cli/2.37.4 we are experiencing heap out memory when triggering those tests via CLI. The tests on the org runs fine, however on the CLI we have:

```
[152:0x7f2dda78b680] 33404559 ms: Mark-Compact 4019.9 (4131.4) -> 4004.8 (4132.1) MB, 1852.90 / 0.00 ms (average mu = 0.145, current mu = 0.026) allocation failure; scavenge might not succeed
[152:0x7f2dda78b680] 33406649 ms: Mark-Compact 4020.7 (4132.1) -> 4005.4 (4132.9) MB, 2021.94 / 0.00 ms (average mu = 0.090, current mu = 0.033) allocation failure; scavenge might not succeed

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
----- Native stack trace -----
/scripts-1089-2040536/step_script: line 259: 152 Aborted (core dumped) sf apex run test -c -v -r junit -w 480 -l RunLocalTests -d output
```

It seems that CLI tries to allocate more then 4GB memory and fails.

Affected commands:

- `sf apex run test -c -v -r junit -w 480 -l RunLocalTests -d output`   - this is complete command we initially found that problem. It runs the tests for many hours and then do some post-processing and finally fail.
- `sf apex get test -c -i 707afas... -d output`  - this is a bit simpler command retrieving existing test run and doing only the post processing. It fails the same way, but it doesn’t  require to run tests, but rather gather existing tests and doing some post processing. This is minimal and convenient way to replicate.

### Steps to reproduce

1. Authorize an org. The credentials are provided within Salesforce case #467133949. There are scratch orgs having our project deployed and tests run. 
2. Find existing test run id on the org. It should begin with `707...`
3. Run the command providing proper test run id `sf apex get test -c -i 7073O00002aQRQqQAO -c -d output` . This usually takes 15-20 min and then it fails with heap out of memory.
4. The replication on 16GB computer does not require additional settings. However it might be that you will have computer with larger amount of RAM it might be required to limit node memory like this `NODE_OPTIONS=--max-old-space-size=4096` (just guessing).

### Expected result

Test report saved into `output` folder.

### Actual result

The CLI exists with non zero exit code and heap out of memory as above.

### System Information

We replicated this on macOS, Linux and Windows. It seems not OS dependent.

```json
{
  "architecture": "darwin-arm64",
  "cliVersion": "@salesforce/cli/2.39.6",
  "nodeVersion": "node-v20.12.2",
  "osVersion": "Darwin 23.4.0",
  "rootPath": "/Users/pawel/.nvm/versions/node/v20.12.2/lib/node_modules/@salesforce/cli",
  "shell": "bash",
  "pluginVersions": [
    "@oclif/plugin-autocomplete 3.0.16 (core)",
    "@oclif/plugin-commands 3.3.1 (core)",
    "@oclif/plugin-help 6.0.21 (core)",
    "@oclif/plugin-not-found 3.1.6 (core)",
    "@oclif/plugin-plugins 5.0.14 (core)",
    "@oclif/plugin-search 1.0.23 (core)",
    "@oclif/plugin-update 4.2.7 (core)",
    "@oclif/plugin-version 2.0.17 (core)",
    "@oclif/plugin-warn-if-update-available 3.0.15 (core)",
    "@oclif/plugin-which 3.1.8 (core)",
    "@salesforce/cli 2.39.6 (core)",
    "apex 3.1.9 (core)",
    "auth 3.6.3 (core)",
    "community 3.0.20 (user)",
    "data 3.3.2 (core)",
    "deploy-retrieve 3.6.6 (core)",
    "dev 2.1.12 (user)",
    "functions 1.22.11 (user)",
    "info 3.2.3 (core)",
    "limits 3.3.4 (core)",
    "marketplace 1.2.4 (core)",
    "org 4.1.3 (core)",
    "packaging 2.4.0 (core)",
    "schema 3.3.4 (core)",
    "settings 2.2.1 (core)",
    "signups 2.0.20 (user)",
    "sobject 1.3.6 (core)",
    "source 3.3.3 (core)",
    "telemetry 3.3.4 (core)",
    "templates 56.2.4 (core)",
    "trust 3.6.6 (core)",
    "user 3.5.2 (core)",
    "vlocityestools 0.24.8 (user)"
  ]
}
```

**Comment by amtrack** [comment]
@pawel-id Maybe you want to switch to use only Aggregated Code Coverage.
That solved the memory issues for us: https://github.com/forcedotcom/salesforcedx-vscode/issues/5554#issuecomment-2096849948

**Comment by daphne-sfdc** [comment]
Issue has been fixed and the code change is released in v61.1.2 of the Salesforce Extensions for VSCode ✅
